### PR TITLE
add authentication to swagger

### PR DESCRIPTION
**Pull request by SergiLomtadze**

### CODE REVIEW

**Comment by SergiLomtadze** [comment]
Problem was in IdentityModel nuget version, I solved it, now it's working as should be

File: src/ExadelMentorship.WebApi/Controllers/LoginController.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    [ApiController]
+    public class LoginController : ControllerBase
+    {
+        private readonly ITokenService _tokenService;
+
+        public LoginController(ITokenService tokenService)
+        {
+            _tokenService = tokenService;
+        }
+        [HttpPost("token/{userName}/{password}")]
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.WebApi/Controllers/LoginController.cs:17]: Pass from body, not query

File: src/ExadelMentorship.WebApi/Token/TokenService .cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+namespace ExadelMentorship.WebApi.Token
+{
+    public class TokenService : ITokenService
+    {
+        private DiscoveryDocumentResponse _discDocument { get; set; }
+        public TokenService()
+        {
+            using (var client = new HttpClient())
+            {
+                _discDocument = client.GetDiscoveryDocumentAsync("https://localhost:7162/.well-known/openid-configuration").Result;
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.WebApi/Token/TokenService .cs:12]: 1. Url to config;
2. Avoid Task.Result

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]

File: src/ExadelMentorship.WebApi/Controllers/LoginController.cs (hunk 1 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.WebApi/Controllers/LoginController.cs]: done

File: src/ExadelMentorship.WebApi/Token/TokenService .cs (hunk 2 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.WebApi/Token/TokenService .cs]: done

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]
Fix the warnings and you are set

**Comment by SergiLomtadze** [comment]
> you are set

done

File: src/ExadelMentorship.Auth/IdentityConfiguration.cs [hunk 3]
```diff
@@ -62,7 +62,7 @@ public class IdentityConfiguration
                 {
                     ClientId = "swagger.client",
                     ClientName = "Swagger Client",
-                    AllowedGrantTypes = GrantTypes.ClientCredentials,
+                    AllowedGrantTypes = GrantTypes.ResourceOwnerPassword,
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.Auth/IdentityConfiguration.cs:65]: Think of how to use Authorization Code + PKCE.

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]

**Comment by SergiLomtadze** [comment]
PR size is failing because of Auth default staff (Quickstart) changes are huge .
I deleted all not needed views.

File: src/ExadelMentorship.Auth/IdentityConfiguration.cs [hunk 4]
```diff
@@ -7,65 +7,63 @@ namespace ExadelMentorship.Auth
...
+        public static IEnumerable<Client> Clients => new[]
+        {
+            new Client
             {
-                new ApiScope("WebApi.read"),
-            };
+                ClientId = "api-swagger",
+                AllowedGrantTypes = GrantTypes.Code,
+                AllowedScopes = { "WebApi.read", "role" },
+                RedirectUris = { "https://localhost:7066/swagger/oauth2-redirect.html" },
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.Auth/IdentityConfiguration.cs:32]: base url to config

File: src/ExadelMentorship.Auth/IdentityConfiguration.cs [hunk 5]
```diff
@@ -7,65 +7,63 @@ namespace ExadelMentorship.Auth
...
+                RequireClientSecret = false,
+                RequirePkce = true,
+                AllowedCorsOrigins = { "https://localhost:7066" },
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.Auth/IdentityConfiguration.cs:35]: same

File: src/ExadelMentorship.WebApi/Program.cs [hunk 6]
```diff
@@ -19,27 +18,59 @@
...
+            },
+            new List<string> { "WebApi.read", "role" }
+        }
+    });
+});
+builder.Services.AddAuthorization();
+builder.Services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+    .AddJwtBearer(JwtBearerDefaults.AuthenticationScheme, o =>
+    {
+        o.Authority = "https://localhost:7046";
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.WebApi/Program.cs:49]: config

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]
Also, fix the PR warnings.

File: src/ExadelMentorship.WebApi/Program.cs (hunk 6 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.WebApi/Program.cs]: done

File: src/ExadelMentorship.Auth/IdentityConfiguration.cs (hunk 5 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.Auth/IdentityConfiguration.cs]: can't use IOptions inside static method

File: src/ExadelMentorship.Auth/IdentityConfiguration.cs (hunk 4 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.Auth/IdentityConfiguration.cs]: can't use IOptions inside static method

**Review by kdavletov-exadel — APPROVED** [review]

**Pull request merged**